src/basic/renderers/select-idp.ts

```diff
@@ -0,0 +1,183 @@
+      }
+    };
+
+    // Add the buttons
+    this.buttons = providers
+      // Only show Local Authentication button when login form is not present
+      .filter((x) => !(this.hasLocalAuthForm && x.provider === 'localAuthentication'))
+      .map((x) => {
+        const handler = (): void => {
+          this.onInput(x.provider);
+        };
+        const element = el<HTMLButtonElement>('button', 'btn btn-social mt-3');
+        let htmlStr;
+
+        if (x.uiConfig) {
+          const { buttonImage, buttonDisplayName } = x.uiConfig;
+          const { baseUrl } = serverConfig;
+          let btnImageUrl;
+
+          if (buttonImage) {
+            btnImageUrl = buttonImage.startsWith('http')
+              ? buttonImage
+              : `${baseUrl}XUI/${buttonImage}`;
+          } else {
+            btnImageUrl = getFallbackImage(x.provider);
+          }
+
+          element.classList.add(`btn-social-${x.provider}`);
+          htmlStr = `
+              ${
+                btnImageUrl
+                  ? `<img height="20px" style="padding-right:5px;" src="${btnImageUrl}" />`
+                  : ``
+              }
+              <span>${buttonDisplayName}</span>
+            `;
+        } else {
+          element.classList.add('btn-outline-secondary');
+          htmlStr = x.provider === 'localAuthentication' ? 'Sign-in with Email' : x.provider;
```

Comment:
Added this ternary in case there's an edge case of a provider other than `localAuthentication` not having a `uiConfig`.